File: backend/app/models/supplier.py
from sqlalchemy import Column, Integer, String, Enum, Text, DateTime
from sqlalchemy.sql import func
from app.database import Base
import enum

class SupplierStatus(str, enum.Enum):
    ACTIVE = "active"
    INACTIVE = "inactive"

class Supplier(Base):
    __tablename__ = "suppliers"

    id = Column(Integer, primary_key=True, index=True)
    code = Column(String(50), unique=True, index=True, nullable=False)
    name = Column(String(100), nullable=False)
    contact_person = Column(String(100), nullable=False)
    email = Column(String(255), nullable=False)
    phone = Column(String(20), nullable=False)
    address = Column(Text, nullable=False)
    status = Column(Enum(SupplierStatus), nullable=False, default=SupplierStatus.ACTIVE)
    notes = Column(Text)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now()) 

File: frontend/app/types/supplier.ts
/**
 * 取引先データの型定義
 */
export interface Supplier {
  id: number;
  name: string;
  code: string;
  contactPerson: string;
  email: string;
  phone: string;
  address: string;
  status: 'active' | 'inactive';
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * 取引先作成用データの型定義
 */
export type CreateSupplierData = Omit<Supplier, 'id' | 'createdAt' | 'updatedAt'>;

/**
 * 取引先更新用データの型定義
 */
export type UpdateSupplierData = Partial<CreateSupplierData>;

/**
 * 取引先ステータスの定義
 */
export const SUPPLIER_STATUSES = {
  active: '取引中',
  inactive: '取引停止',
} as const; 